src/modules/rlm_files/rlm_files.c

```diff
@@ -104,17 +108,37 @@ static int pairlist_cmp(void const *a, void const *b)
 	return fr_value_box_cmp(((PAIR_LIST const *)a)->name, ((PAIR_LIST const *)b)->name);
 }
 
-static int getusersfile(TALLOC_CTX *ctx, char const *filename, rbtree_t **ptree)
+static PAIR_LIST *find_entry(rlm_files_t const *inst, fr_value_box_t *key, rbtree_t *ptree, fr_trie_t const *ptrie)
+{
+	PAIR_LIST my_e;
+
+	switch (inst->key_data_type) {
+	case FR_TYPE_IPV4_ADDR:
```

Comment:
Can you just add an enum which describes which tree is in use, and then set the value in instantiate.

- Makes it easier to add new lookup types in future, or mix up the logic which determines which tree is in use.
- The same four types aren't copy/pasted around multiple areas of the code.
- It's faster.